genomeuploader/ena.py

```
@@ -32,265 +36,396 @@ class NoDataException(ValueError):
+def query_scientific_name(scientific_name, search_rank=False):
+    url = "https://www.ebi.ac.uk/ena/taxonomy/rest/scientific-name/{}".format(scientific_name)
+    response = requests.get(url)
 
-    def get_study_runs(self, study_acc, webin, password, fields=None, search_params=None):
-        data = self.get_default_params()
-        data['result'] = 'read_run'
-        data['fields'] = fields or RUN_DEFAULT_FIELDS
-        data['query'] = '(study_accession=\"{}\" OR secondary_study_accession=\"{}\")'.format(study_acc, study_acc)
+    try:
+        # Will raise exception if response status code is non-200
+        response.raise_for_status()
+    except requests.exceptions.HTTPError:
+        if search_rank:
+            return False, "", ""
+        else:
+            return False, ""
 
-        if search_params:
-            data.update(search_params)
+    try:
+        res = response.json()[0]
+    except IndexError:
+        if search_rank:
+            return False, "", ""
+        else:
+            return False, ""
+
+    submittable = res.get("submittable", "").lower() == "true"
+    taxid = res.get("taxId", "")
+    rank = res.get("rank", "")
+
+    if search_rank:
+        return submittable, taxid, rank
+    else:
+        return submittable, taxid
+
+
+class EnaQuery:
+    def __init__(self, accession, query_type, private=False):
+        self.private_url = "https://www.ebi.ac.uk/ena/submit/report"
+        self.public_url = "https://www.ebi.ac.uk/ena/portal/api/search"
+        self.browser_url = "https://www.ebi.ac.uk/ena/browser/api/xml"
+        self.accession = accession
+        self.acc_type = parse_accession(accession)
+        username, password = configure_credentials()
+        if username is None or password is None:
+            logging.error("ENA_WEBIN and ENA_WEBIN_PASSWORD are not set")
+        if username and password:
+            self.auth = (username, password)
+        else:
+            self.auth = None
```

We always need the credentials for submission - we might remove them when querying ENA publicly, but we'll need to raise an error if no credentials are available.